src/destructors.md

```diff
@@ -2,9 +2,7 @@
 
 When an [initialized]&#32;[variable] in Rust goes out of scope or a [temporary]
 is no longer needed its _destructor_ is run. [Assignment] also runs the
-destructor of its left-hand operand, unless it's an uninitialized variable. If a
-[struct] variable has been partially initialized, only its initialized fields
-are dropped.
```

Comment:
We can still do:
```rust
struct ND(u8);

impl Drop for ND {
    fn drop(&mut self) {
        dbg!(self.0);
    }
}

struct Prod(ND, ND, ND);

fn main() {
    let mut prod = Prod(ND(0), ND(1), ND(2));
    core::mem::forget(prod.0);
    // prod.0 is not initialized here.
}
```

===>

```rust
[src/main.rs:5] self.0 = 1
[src/main.rs:5] self.0 = 2
```

Reply:
OK.  So how about just updating the example to do that? I just pushed a change, with the wording from #514.

Reply:
Or is a "partial move" different from a partially initialized value?

Reply:
It seems to me that "partial move" is something that happens and "partially initialized" is a state of being that can result from a "partial move".

cc @RalfJung @matthewjasper

Reply:
The corresponding adjective would be "partially moved out of", I guess.

But as a starter, I don't see how removing `struct` here helps with anything. If we (re)spin "partially initialized" to mean "partially moved out of", this is still about structs (and tuples), like before.

However, I think it would be better to change the wording. or do we consistently call moved-out-of variables "uninitialized"?

Reply:
> However, I think it would be better to change the wording. or do we consistently call moved-out-of variables "uninitialized"?

I'm not sure. I'm not too keen on a particular wording as long as the content is there (which it is).

Reply:
I only removed the word "struct" to avoid confusion (and that's what #514 did), since it could be a struct, enum, tuple, or whatever.  Since the example uses a tuple, it seemed better to not be so specific.

I think it would be good to be clearer about this, but I also think #514 would be the place to discuss and debate the specifics. I'd like to keep this PR focused on the minimum to get the tests to pass.  Does the new example make sense? Should it use a different wording? Or should it just be removed?

Reply:
> I'd like to keep this PR focused on the minimum to get the tests to pass.

Then please don't make unrelated wording changes. :)

Reply:
Fair, I removed it.